tsfc/kernel_interface/firedrake_loopy.py

```diff
@@ -153,12 +153,14 @@ def register_requirements(self, ir):
         provided by the kernel interface."""
         self.oriented, self.cell_sizes, self.tabulations = check_requirements(ir)
 
-    def construct_kernel(self, return_arg, impero_c, index_names):
+    def construct_kernel(self, return_arg, impero_c, index_names, first_coefficient_fake_coords):
         """Constructs an :class:`ExpressionKernel`.
 
         :arg return_arg: loopy.GlobalArg for the return value
         :arg impero_c: gem.ImperoC object that represents the kernel
         :arg index_names: pre-assigned index names
+        :arg first_coefficient_fake_coords: If true, the kernel's first
+            coefficient is a constructed UFL coordinate field
```

Comment:
TBH, this is a horrible variable name. I think what this means is that the kernel expects to be fed:

```
[mesh.coordinates] + extract_coefficients(expression)
```

Is that right?

Reply:
Didn't see this comment in time. Yes that is correct

Reply:
If you have a suggested better interface and/or variable name to say and I can make a new PR

Reply:
I think in various places in Firedrake we have similar variables to this e.g. for cell sizes. They are called something like `needs_cell_sizes`, maybe you just want to rename to `needs_coords`.